src/appsync.ts

```diff
@@ -180,391 +193,385 @@ export class AppsyncResolver<
-      appsync.BaseResolverProps,
-      "typeName" | "fieldName" | "cachingConfig"
-    >
-  ): SynthesizedAppsyncResolver {
-    const fields = this.getResolvableFields(api);
-
-    return new SynthesizedAppsyncResolver(
-      api,
-      `${options.typeName}${options.fieldName}Resolver`,
-      {
-        ...options,
-        api,
-        templates: fields.templates,
-        dataSource: fields.dataSource,
-        requestMappingTemplate: fields.requestMappingTemplate,
-        responseMappingTemplate: fields.responseMappingTemplate,
-        pipelineConfig: fields.pipelineConfig,
-      }
-    );
+  constructor(
+    scope: Construct,
+    id: string,
+    props: AppsyncResolverProps,
+    resolve: ResolverFunction<Arguments, Result, Source>
+  ) {
+    super(scope, id);
+    this.decl = validateFunctionDecl(resolve, "AppsyncResolver");
+
+    this.resolvers = memoize(() => synthResolvers(props.api, this.decl));
+
+    this.resource = new aws_appsync.CfnResolver(this, "Resource", {
+      apiId: props.api.apiId,
+      typeName: props.typeName,
+      fieldName: props.fieldName,
+      kind: Lazy.string({
+        produce: () => {
+          const { pipelineConfig } = this.resolvers();
+          return pipelineConfig === undefined || pipelineConfig.length === 0
+            ? "UNIT"
+            : "PIPELINE";
+        },
+      }),
+      requestMappingTemplate: Lazy.string({
+        produce: () => this.resolvers().requestMappingTemplate.renderTemplate(),
+      }),
+      pipelineConfig: Lazy.any({
+        produce: () => {
+          const { pipelineConfig } = this.resolvers();
+          if (pipelineConfig === undefined || pipelineConfig.length === 0) {
+            return undefined;
+          } else {
+            return {
+              functions: pipelineConfig.map((func) => func.functionId),
+            };
+          }
+        },
+      }),
+      responseMappingTemplate: Lazy.string({
+        produce: () =>
+          this.resolvers().responseMappingTemplate.renderTemplate(),
+      }),
+    });
   }
+}
 
+export interface AppsyncFieldOptions
+  extends Omit<
+    appsync.ResolvableFieldOptions,
+    keyof ReturnType<typeof synthResolvers>
+  > {
+  api: appsync.GraphqlApi;
+}
+
+/**
+ * Generate a resolvable field to use with AppSync CDK's Code field strategy.
+ *
+ * ```ts
+ * import * as appsync from "@aws-cdk/aws-appsync-alpha";
+ *
+ * const api = new appsync.GraphQLApi(..);
+ *
+ * ```ts
+ * const getPerson = new AppsyncResolver<{id: string}, Person | undefined>(
+ *   ($context, id) => {
+ *     const person = table.get({
+ *       key: {
+ *         id: $util.toDynamoDB(id)
+ *       }
+ *     });
+ *     return person;
+ *   });
+ * ```
+ *
+ * // code first person type
+ * const personType = api.addType(appsync.ObjectType(...));
+ *
+ * api.addQuery("getPerson", getPerson.getField(api, personType))
+ * getPerson.createResolver(api, {
+ *   typeName: "Query",
+ *   fieldName: "getPerson"
+ * });
+ * ```
```

Comment:
Oops, these are not updated. That reminds me - i need to update the website docs too